File: split/scanspec.pas
procedure scanspec(c: groupcode; threecodes: boolean);
label
  40;
var
  s: integer;
  speccode: 0..1;
begin
  if threecodes then
    s := savestack[saveptr + 0].int;
  if scankeyword(841) then
    speccode := 0
  else if scankeyword(842) then
    speccode := 1
  else
  begin
    speccode := 1;
    curval := 0;
    goto 40;
  end;
  scandimen(false, false, false);
  40:
    if threecodes then
    begin
      savestack[saveptr + 0].int := s;
      saveptr := saveptr + 1;
    end;
  savestack[saveptr + 0].int := speccode;
  savestack[saveptr + 1].int := curval;
  saveptr := saveptr + 2;
  newsavelevel(c);
  scanleftbrace;
end;

File: cpp/scanspec.cpp
#include "scanspec.h"
#include "scandimen.h"
#include "scankeyword.h"
#include "newsavelevel.h"
#include "scanleftbrace.h"

void scanspec(groupcode c, bool threecodes)
{
	int s;
	if (threecodes)
		s = savestack[saveptr+0].int_;
	int speccode;
	if (scankeyword(841)) //to
	{
		speccode = 0;
		scandimen(false, false, false);
	}
	else 
		if (scankeyword(842)) //spread
		{
			speccode = 1;
			scandimen(false, false, false);
		}
		else
		{
			speccode = 1;
			curval = 0;
		}
	if (threecodes)
		savestack[saveptr++].int_ = s;
	savestack[saveptr].int_ = speccode;
	savestack[saveptr].int_ = curval;
	saveptr += 2;
	newsavelevel(c);
	scanleftbrace();
}
